comment/comment_revision_repository_blackbox_test.go

```diff
@@ -0,0 +1,128 @@
+	clean              func()
+	testIdentity1      account.Identity
+	testIdentity2      account.Identity
+	testIdentity3      account.Identity
+}
+
+// SetupSuite overrides the DBTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (s *revisionRepositoryBlackBoxTest) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+	// Make sure the database is populated with the correct types (e.g. bug etc.)
+	if _, c := os.LookupEnv(resource.Database); c != false {
+		if err := models.Transactional(s.DB, func(tx *gorm.DB) error {
+			return migration.PopulateCommonTypes(context.Background(), tx, workitem.NewWorkItemTypeRepository(tx))
+		}); err != nil {
+			panic(err.Error())
+		}
+	}
+}
+
+func (s *revisionRepositoryBlackBoxTest) SetupTest() {
+	s.repository = comment.NewRepository(s.DB)
+	s.revisionRepository = comment.NewRevisionRepository(s.DB)
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+	testIdentity1, err := testsupport.CreateTestIdentity(s.DB, "jdoe1", "test")
+	require.Nil(s.T(), err)
+	s.testIdentity1 = testIdentity1
+	testIdentity2, err := testsupport.CreateTestIdentity(s.DB, "jdoe2", "test")
+	require.Nil(s.T(), err)
+	s.testIdentity2 = testIdentity2
+	testIdentity3, err := testsupport.CreateTestIdentity(s.DB, "jdoe3", "test")
+	require.Nil(s.T(), err)
+	s.testIdentity3 = testIdentity3
+}
+
+func (s *revisionRepositoryBlackBoxTest) TearDownTest() {
+	//s.clean()
+}
+
+func (s *revisionRepositoryBlackBoxTest) TestStoreCommentRevisions() {
+	// given
+	// create a comment
+	c := newComment("A", "Body", rendering.SystemMarkupMarkdown)
+	err := s.repository.Create(context.Background(), c, s.testIdentity1.ID)
+	require.Nil(s.T(), err)
+	// modify the comment
+	c.Body = "Updated body"
+	c.Markup = rendering.SystemMarkupPlainText
+	err = s.repository.Save(
+		context.Background(), c, s.testIdentity2.ID)
+	require.Nil(s.T(), err)
+	// modify again the comment
+	c.Body = "Updated body2"
+	c.Markup = rendering.SystemMarkupMarkdown
+	err = s.repository.Save(context.Background(), c, s.testIdentity2.ID)
+	require.Nil(s.T(), err)
+	// delete the comment
+	err = s.repository.Delete(
+		context.Background(), c.ID, s.testIdentity3.ID)
+	require.Nil(s.T(), err)
+	// when
+	commentRevisions, err := s.revisionRepository.List(context.Background(), c.ID)
+	// then
+	require.Nil(s.T(), err)
+	require.Len(s.T(), commentRevisions, 4)
+	// revision 1
+	revision1 := commentRevisions[0]
+	assert.Equal(s.T(), c.ID, revision1.CommentID)
+	assert.Equal(s.T(), comment.RevisionTypeCreate, revision1.Type)
+	assert.Equal(s.T(), "Body", *revision1.CommentBody)
+	assert.Equal(s.T(), rendering.SystemMarkupMarkdown, *revision1.CommentMarkup)
+	assert.Equal(s.T(), s.testIdentity1.ID, revision1.ModifierIdentity)
+	// revision 2
+	revision2 := commentRevisions[1]
```

Comment:
I'm torn between 

Whether we should move repetitive code inside a different method ,
and 
the prospect of losing line numbers in case of error.  

:)